### PR TITLE
Upgrade project to use go 1.13 

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
Upgrade the project to work/used 1.13 go and its tests

**Motivation**
Closes: https://github.com/kubernetes-sigs/kubebuilder/issues/1134

### CODE REVIEW

**Comment by camilamacedo86** [comment]
c/c @mengqiy

**Comment by camilamacedo86** [comment]
HI @mengqiy, 

Could you please help us move forward with by doing its review?    
Really tks for your help and guidance. 

Also, note that we have https://github.com/kubernetes-sigs/kubebuilder/issues/1098 to refactory . the scripts which will be the next task that I would do after we merge it.

File: common.sh [hunk 1]
```diff
@@ -144,7 +143,8 @@ function build_kb {
   GO111MODULE=on go build $opts -o $tmp_root/kubebuilder/bin/kubebuilder ./cmd
 }
 
-function fetch_go_tools {
+# Required to install the go dep in the https://prow.k8s.io/.
+function install_dep_by_git {
```
Comment (mengqiy) [inline common.sh]: Why we use 2 different approaches to install dep?

File: .travis.yml [hunk 2]
```diff
@@ -10,23 +10,23 @@ os:
   - osx
 
 go:
-- "1.12"
+- "1.13"
```
Comment (mengqiy) [inline .travis.yml]: Please revert this change.

File: testdata/project-v2/go.mod [hunk 3]
```diff
@@ -1,12 +1,11 @@
 module sigs.k8s.io/kubebuilder/testdata/project-v2
 
-go 1.12
+go 1.13
```
Comment (mengqiy) [inline testdata/project-v2/go.mod]: Please revert this change.

File: common.sh (hunk 1 above)
Comment (camilamacedo86) [reply common.sh]: Because after the upgrade to use go version 1.13 these steps are not working after for the CI tests and prow will face the GOBIN issue.

File: .travis.yml (hunk 2 above)
Comment (camilamacedo86) [reply .travis.yml]: But the goal is to upgrade the project to work/use go version 1.13. So, why the Travis would test check it with 1.12?

File: testdata/project-v2/go.mod (hunk 3 above)
Comment (camilamacedo86) [reply testdata/project-v2/go.mod]: ^ the same

**Comment by camilamacedo86** [comment]
Hi @mengqiy, 

Just to register here, I spoke with @DirectXMan12 as you recommend about it. 
Now we are waiting for check if we can move forward with now or should wait a little more since the k8s still using 1.12.

File: common.sh (hunk 1 above)
Comment (camilamacedo86) [reply common.sh]: PS.: Note that it shows that will be removed in the future versions as well, when we no longer support dep.

File: testdata/project-v2/go.mod (hunk 3 above)
Comment (DirectXMan12) [reply testdata/project-v2/go.mod]: you need to update the scaffolding file too

Comment (camilamacedo86) [reply testdata/project-v2/go.mod]: Done @DirectXMan12 :-) 
Thank you for the good spot 🥇

**Comment by camilamacedo86** [comment]
H @DirectXMan12, 

All done 👍 Also, see [here](https://github.com/kubernetes/kubernetes/pull/82809) that shows that we can upgrade safely .

**Comment by mengqiy** [comment]
k/k is now on golang1.13 with https://github.com/kubernetes/kubernetes/pull/82809 merged.

File: common.sh (hunk 1 above)
Comment (mengqiy) [reply common.sh]: Can you please describe the GOBIN issue with prow?
I'd suggest move put some description in the comments.

**Comment by camilamacedo86** [comment]
Hi @mengqiy and @DirectXMan12,

Let's merge this one? If not, let me know what is missing for we are able to move forward.

**Comment by DirectXMan12** [comment]
> k/k is now on golang1.13 with kubernetes/kubernetes#82809 merged.

Full steam ahead on the 1.13 train, then :steam_locomotive:

File: docs/book/src/cronjob-tutorial/testdata/project/go.mod [hunk 4]
```diff
@@ -1,11 +1,10 @@
 module tutorial.kubebuilder.io/project
 
-go 1.12
+go 1.13
 
 require (
 	github.com/go-logr/logr v0.1.0
 	github.com/robfig/cron v1.2.0
-	golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2 // indirect
```
Comment (Adirio) [inline docs/book/src/cronjob-tutorial/testdata/project/go.mod]: My IDE is inserting this line automatically when I pulled this PR. Does it really need to be removed?

File: testdata/project-v2/go.mod [hunk 5]
```diff
@@ -1,12 +1,11 @@
...
 
 require (
 	github.com/go-logr/logr v0.1.0
 	github.com/onsi/ginkgo v1.6.0
 	github.com/onsi/gomega v1.4.2
-	golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2 // indirect
```
Comment (Adirio) [inline testdata/project-v2/go.mod]: Same as above.

**Comment by Adirio** [comment]
@camilamacedo86 @DirectXMan12 @mengqiy @droot

File: docs/book/src/cronjob-tutorial/testdata/project/go.mod (hunk 4 above)
Comment (camilamacedo86) [reply docs/book/src/cronjob-tutorial/testdata/project/go.mod]: It will be removed when we do a go mod tidy than it should not be committed.
